app.ts

```diff
@@ -0,0 +1,341 @@
+    entry[1].removeEventListener("input", validate);
+    entry[1].parentNode.removeChild(entry[1]);
+  });
+
+  //removing hide class (to see the actualindex content)
+  let removeHideClass: object = callElements(index[0], "value_");
+  (<any>Object).entries(removeHideClass).forEach(<T>(entry: T) => {
+    entry[1].classList.remove("hide");
+  });
+
+  //removing save and canccel button and event listeners
+  let savebtn: HTMLElement = document.getElementById(`save_${index[0]}`);
+  let cancelbtn: HTMLElement = document.getElementById(`cancel_${index[0]}`);
+  savebtn.removeEventListener("click", save);
+  cancelbtn.removeEventListener("click", cancel);
+  //getting back the edit button
+  document.getElementById(
+    `edit_wrapper_${index[0]}`
+  ).innerHTML = `<button class="btn btn-secondary" id="edit_${
+    index[0]
+  }">edit</button>`;
+
+  let editbtn: HTMLElement = document.getElementById(`edit_${index}`);
+  editbtn.addEventListener("click", editRow);
+};
+
+var save = (e: any): void => {
+  // if(validateform(index)) {
+  //     alert("Check the values in the form.")
+  // } else {
+
+  let index: object = /\d+/.exec(e.path[0].id);
+
+  document.getElementById(e.path[0].id).removeEventListener("click", deleteRow);
+  let elements: object = callElements(index[0], "input_");
+  let { fname, mname, lname, email, number, address } = <UserHtmlElements>(
+    elements
+  );
+
+  //removing input fields and event listeners
+  (<any>Object).entries(elements).forEach(<T>(entry: T) => {
+    entry[1].removeEventListener("input", validate);
+    entry[1].parentNode.removeChild(entry[1]);
+  });
+
+  //getting back the hidden data
+  let removeHideClass: object = callElements(index[0], "value_");
+
+  (<any>Object).entries(removeHideClass).forEach(<T>(entry: T) => {
+    entry[1].classList.remove("hide");
+  });
+
+  let {
+    fname: resetfname,
+    lname: resetlname,
+    mname: resetmname,
+    email: resetemail,
+    number: resetnumber,
+    address: resetaddress
+  } = <UserHtmlElements>removeHideClass;
+  //replacing the value with new values on save
+  resetfname.innerHTML = fname.value;
+  resetlname.innerHTML = lname.value;
+  resetmname.innerHTML = mname.value;
+  resetemail.innerHTML = email.value;
+  resetnumber.innerHTML = number.value;
+  resetaddress.innerHTML = address.value;
+
+  //modify the data
+  data[index[0]]["First Name"] = fname.value;
+  data[index[0]]["Last Name"] = lname.value;
+  data[index[0]]["Middle Name"] = mname.value;
+  data[index[0]]["Email"] = email.value;
+  data[index[0]]["Phone Number"] = number.value;
+  data[index[0]]["Address"] = address.value;
+
+  //finally getting the edit button back
+  document.getElementById(
+    `edit_wrapper_${index}`
+  ).innerHTML = `<button id="edit_${index}" class="btn btn-secondary" >edit</button>`;
+  document.getElementById(`edit_${index}`).addEventListener("click", editRow);
+  // }
+};
+
+var makeLayout = (items: object[]): string[] => {
```

Comment:
why object ?

Reply:
Because I am passing an array of objects as it's first parameter.